docs/RFCS/remove_sql_cli.rst

```diff
@@ -0,0 +1,100 @@
+:Feature Name: Supersede the SQL command-line client with psql
```

Comment:
Please convert this to markdown; as I said in #4056 I don't think we need two different formats for RFCs.